lib/_http_client.js

```diff
@@ -68,6 +68,10 @@ function ClientRequest(options, cb) {
   self.socketPath = options.socketPath;
   self.timeout = options.timeout;
 
+  const _method = options.method;
+  if (_method != null && typeof _method !== 'string') {
+    throw new TypeError('Method must be a string');
+  }
   var method = self.method = (options.method || 'GET').toUpperCase();
```

Comment:
We should probably use `options.method` or `_method` consistently. Right now it's a mix.